backend/src/main/java/fk/prof/backend/Configuration.java

```diff
@@ -0,0 +1,401 @@
+package fk.prof.backend;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import fk.prof.backend.leader.election.KillBehavior;
+import io.vertx.core.DeploymentOptions;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.core.json.JsonObject;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.util.Map;
+
+/**
+ * Created by gaurav.ashok on 21/05/17.
+ */
+public class Configuration {
+    @JsonCreator
+    public Configuration(@JsonProperty("ip.address") String ipAddress,
```

Comment:
I kept all the member final and so they can only be initialized from constructor.
Pro for this is that all file size will be smaller, all string names will present alongside be other.
Any defaulting will be present in constructor.